File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public static int[] twoSum(int[] nums, int target) {
        HashMap <Integer, Integer> work = new HashMap<>();
        for (int i=0; i<nums.length; i++) {
            int element = nums[i];
            int complement = target - element;
            if (work.containsKey(complement)) {
                int[] result = {work.get(complement), i};
                return result;
            } else {
                work.put(element, i);
            }
        }
        return null;
    }
    public static void main(String[] args) {
        int[] example = {2,7,11,15};
        System.out.println(Arrays.toString(twoSum(example, 9)));
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) -1

        while l < r:
            curr = numbers[l] + numbers[r]

            if curr > target:
                r -= 1
            elif curr <  target:
                l += 1
            else:
                return [1+l, r+1]